test/acceptance/features/steps/kind.py

```diff
@@ -118,17 +118,77 @@ def __load_image(self, img: str):
+        """
+        Deploys the Service Agent into a cluster's namespace
+        """
+        img = "agentsvc:latest"
+        kubeconfig = self.cluster_provisioner.kubeconfig()
+        with tempfile.NamedTemporaryFile(prefix=f"kubeconfig-{self.cluster_name}-") as t:
+            t.write(kubeconfig.encode("utf-8"))
+            self.__install_crd_and_build_svc_image(t.name, img)
+            self.__load_agentsvc_image()
+            self.__deploy_agentsvc(t.name, img, namespace)
+
+    def deploy_agentapp(self, namespace: str):
+        """
+        Deploys Application Agent into a cluster's namespace
+        """
+        img = "agentapp:latest"
+        kubeconfig = self.cluster_provisioner.kubeconfig()
+        with tempfile.NamedTemporaryFile(prefix=f"kubeconfig-{self.cluster_name}-") as t:
+            t.write(kubeconfig.encode("utf-8"))
+            self.__install_crd_and_build_app_image(t.name, img)
+            self.__load_agentapp_image()
+            self.__deploy_agentapp(t.name, img, namespace)
+
+    def __install_crd_and_build_app_image(self, kubeconfig_path: str, img: str):
+        out, err = self.__build_install_base_cmd(kubeconfig_path, img).run("make agentapp install docker-build")
+        print(out)
+        assert err == 0, "error installing manifests and building agent app  controller"
+
+    def __install_crd_and_build_svc_image(self, kubeconfig_path: str, img: str):
+        out, err = self.__build_install_base_cmd(kubeconfig_path, img).run("make agentsvc install docker-build")
+        print(out)
+        assert err == 0, "error installing manifests and building agent app  controller"
+
+    def __deploy_agentapp(self, kubeconfig_path: str, img: str, namespace: str):
+        out, err = self.__build_install_base_cmd(kubeconfig_path, img).setenv("NAMESPACE", namespace).run("make agentapp deploy")
+        print(out)
+        assert err == 0, f"error deploying Agent app's controller into cluster {self.cluster_name}"
+
+    def __deploy_agentsvc(self, kubeconfig_path: str, img: str, namespace: str):
+        out, err = self.__build_install_base_cmd(kubeconfig_path, img).setenv("NAMESPACE", namespace).run("make agentsvc deploy")
+        print(out)
+        assert err == 0, f"error deploying Agent app's controller into cluster {self.cluster_name}"
+
+    def __load_agentapp_image(self):
+        cmd = f'make agentapp docker-build && kind load docker-image --name {self.cluster_name} $IMG'
+        output, exit_code = Command().setenv("IMG", "agentapp:latest").run(cmd)
+        print(output)
+
+        if exit_code != 0:
+            raise Exception("error loading agentapp image")
+
+    def __load_agentsvc_image(self):
+        cmd = f'make agentsvc docker-build && kind load docker-image --name {self.cluster_name} $IMG'
+        output, exit_code = Command().setenv("IMG", "agentsvc:latest").run(cmd)
+        print(output)
+
+        if exit_code != 0:
+            raise Exception("error loading agentsvc image")
```

Comment:
I propose the following approach (also related to next comment)

```suggestion
    def __load_image(self, image: str):
        cmd = f' load docker-image --name {self.cluster_name} $IMG'
        output, exit_code = Command().setenv("IMG", img).run(cmd)
        print(output)

        if exit_code != 0:
            raise Exception(f"error loading image {image}")
```